### ISSUE TITLE
[LFRic] Support for Wchi function space

### ISSUE THREAD

**Issue by TeranIvy** [stfc/PSyclone#539]
Whilst reviewing PR #532, @rupertford asked a very good question - why are we not supporting `Wchi` space? 
After some exploration of the LFRic trunk, it seems that `Wchi` is not actually used. The coordinate fields (`chi`) metadata are defined as `ANY_SPACE_9` in the kernels. `chi_space` is set  to `W0` in `runtime_constant_alg_mod.x90` if  `coordinate_order == 0` and to `Wchi` otherwise and we are running with order 0 on trunk.

So, my questions to @thomasmelvin are:
- Are we planning to use `Wchi` at some point?
- If we use `Wchi`, should metadata for `chi` fields in kernels then be `Wchi` and not `ANY_SPACE_9`?

**Comment by thomasmelvin** [comment]
The reason chi always uses ANY_SPACE_9 is, as you point out, that we have option of chi being in either W0 (continuous) or Wchi (discontinuous) and so we needed to cover both basis. The reason for keeping W0 was that it requires less storage than the equivalent Wchi space.

However the work that Stephen P is doing (using cubed-sphere coordinates) means that we would need a discontinuous chi field and hence need to use Wchi at which point we should change the kernel metadata to support it.

In summary it would be good if psyclone supports Wchi as a discontinuous field and we should look at removing the option to use W0 for chi and change the metadata from ANY_SPACE_9 to Wchi  in LFRic
